changes/2025-01-17_key-store-admin/background.md

```diff
@@ -0,0 +1,230 @@
+
+## Key Management
+
+`Key Management` is a union,
+such that additional options may be added at a later date.
+
+Members of `Key Management` are elements that
+authorize the use of Branch Key Cryptographic Materials
+by decrypting them in a manner that
+authenticates all attributes of the Branch Key.
+
+They are also used to create Branch Key Encrypted Cryptographic Materials,
+and when doing so MUST cryptographically bind all attributes of the Branch Key
+to the cipher-text.
+
+Currently, there is only one `Key Management` option.
+
+`Key Management` is never used directly by the Key Store Admin.
+
+However, it was introduced into the Key Store
+as part of the [Key Store Persistence Changes](../../changes/2024-6-17_key-store-persistance/background.md).
+It is, at least at this time,
+not visible to any MPL Consumer facing operations.
+
+### `AwsKms`
+
+`AwsKms` is the only `Key Management` option currently
+supported by the Key Store or Key Store Admin.
+
+This option is provided by default,
+if no other `Key Management` is provided.
+
+`AwsKms` represents AWS Key Management Service (KMS).
+
+The structure has two optional fields:
+
+- `grantTokenList`, a list of strings
+- `kmsClient`, a reference to an AWS SDK KMS client
+
+If no `grantTokenList` is provided,
+it defaults to an empty list.
+
+If no `kmsClient` is provided,
+the default KMS client is constructed via the AWS SDK.
+
+## Key Management Strategy
+
+`Key Management Strategy` is a union,
+such that additional options may be added at a later date.
+
+`Key Management Strategy` determines which Operations
+of a `Key Management` are used by the Client.
+
+For example,
+an MPL Consumer MAY want to avoid calling `kms:ReEncrypt`,
+and would rather use `kms:Decrypt` followed by `kms:Encrypt`
+to re-wrap a new DECRYPT_ONLY Branch Key as a ACTIVE Branch Key.
+
+At this time,
+there are only two
+`Key Management Strategy`s.
+
+### AWS KMS ReEncrypt (default)
+
+`AwsKmsReEncrypt` dictates the Key Store Operation use
+AWS KMS' ReEncrypt Operation to
+[authenticate a Key Store Item](../../framework/branch-key-store.md#authenticating-a-keystore-item)
+or re-wrap Branch Keys
+during [Wrapped Branch Key Creation](../../framework/branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsReEncrypt` is a structure that holds a `AwsKms`,
+which MAY contain the KMS Client
+or Grant Tokens the Key Store,
+if they are set,
+will use when calling KMS.
+
+`AwsKmsReEncrypt` is the default option if
+`Key Management Strategy` is marked as optional
+and the parameter is unfilled.
+
+### AWS KMS Decrypt Encrypt
+
+`AwsKmsDecryptEncrypt` dictates the Key Store Operation to use
+AWS KMS' Decrypt Operation followed by AWS KMS Encrypt Operation
+to re-wrap Branch Keys
+during [Wrapped Branch Key Creation](../../framework/branch-key-store.md#wrapped-branch-key-creation).
+
+`AwsKmsDecryptEncrypt` is a structure that holds two `AwsKms`,
+one designated for Decrypt,
+one designated for Encrypt.
```

Comment:
```suggestion
`AwsKmsDecryptEncrypt` is a structure that holds two `AwsKms` clients,
one for Decrypt, and one for Encrypt.
```

Reply:
Hmm... I like the term `designated`,
as later we will use the `Encrypt` client for GenerateDataKey and Decrypt.

Maybe I will change the wording to "Primarily but not exclusively for"